core/node/notifications/auth.go

```diff
@@ -136,8 +139,9 @@ func (s *Service) FinishAuthentication(
 
 	// make sure that the caller has access to the private key from which user id was derived
 	chal := raw.(*authenticationChallenge)
-	if !chal.Verify(challenge, msg.GetSignature(), msg.GetDelegateSig(), msg.GetDelegateExpiryEpochMs()) {
-		return nil, RiverError(Err_PERMISSION_DENIED, "bad signature").Tag("user", userID)
+	err := chal.Verify(ctx, challenge, msg.GetSignature(), msg.GetDelegateSig(), msg.GetDelegateExpiryEpochMs())
+	if err != nil {
+		return nil, RiverError(Err_PERMISSION_DENIED, "bad signature").Tag("user", userID).Tag("error", err)
```

Comment:
I don't think that's a good idea because clients will making this request and need handle errors in the response. 
The api should provide a consistent error message for the case where the challenge failed to verify, so that we can properly handle it and accurately retry or show an error to the user. 
If we're just forwarding along an error that came from a random library we lose clarity for the client implementation and set ourselves up for breaking changes if the library gets refactored or replaced